packages/blade/src/components/ActionList/ActionListBox.web.tsx

```diff
@@ -36,6 +44,120 @@ const _ActionListBox = React.forwardRef<HTMLDivElement, ActionListBoxProps>(
   },
 );
 
-const ActionListBox = assignWithoutSideEffects(_ActionListBox, { displayName: 'ActionListBox' });
+const ActionListBox = assignWithoutSideEffects(React.memo(_ActionListBox), {
+  displayName: 'ActionListBox',
+});
 
-export { ActionListBox };
+const getVirtualItemParams = ({
+  theme,
+  isMobile,
+}: {
+  theme: Theme;
+  isMobile: boolean;
+}): {
+  itemHeight: number;
+  actionListBoxHeight: number;
+} => {
+  const itemHeightResponsive = getItemHeight(theme);
+  const actionListPadding = getActionListPadding(theme);
+  const actionListBoxHeight = actionListMaxHeight - actionListPadding * 2;
+
+  return {
+    itemHeight: isMobile
+      ? itemHeightResponsive.itemHeightMobile
+      : itemHeightResponsive.itemHeightDesktop,
+    actionListBoxHeight,
+  };
+};
+
+const useFilteredItems = (
```

Comment:
Can we add 1 line comment on "what" is this doing?

Reply:
done